lib/activity.rb

```diff
@@ -0,0 +1,34 @@
+class Activity
+
+  attr_reader :name,
+              :participants,
+              :total_cost,
+              :owed
+
+  def initialize(name)
+    @name = name
+    @participants = Hash.new(0)
+  end
+
+  def add_participant(name, paid)
+    @participants[name] += paid
+  end
+
+  def total_cost
+    @participants.map do |person,paid|
+        paid
+    end.flatten.sum
+  end
+
+  def split
+    self.total_cost / @participants.count
```

Comment:
This totally works, but you do not _have_ to call total cost on self, you could just call total_cost - `total_cost / @participants.count`